playbooks/roles/deploy-osh/tasks/kubectl-install.yml

```diff
@@ -61,12 +63,20 @@
       stat:
         path: "{{ kubeconfig_file_path }}"
       delegate_to: localhost
+      register: new_config
+
+    - name: Find if an existing .kube/config file
```

Comment:
Could you explain why and spin that up in a new commit?

Reply:
This is we already have in ucp branch and have made life quite easier as now no need to refresh kube config every 2nd day in long running caasp cluster
So what's the concern other than part of a branch.

Reply:
I am confused, it seems exactly like https://github.com/SUSE-Cloud/socok8s/pull/115/files#diff-46a36719526ac2ed40c3a997043b868dR35 . 

Do you mean when you don't use automation, you still want to test the age of the kubeconfig, and have a way to copy the file?

In that case, please create a new PR, and update the logic I stated above.
It should updated like this to avoid repeating ourselves:
1) Check if kubeconfig is recent
2) If automation is present, update the kubeconfig with velum-interactions
3) If not present, pdate the kubeconfig by copying a local file (so about the same logic as now, just reusing the stat in 1) and overriding the file.

Still worth doing in another PR.

Reply:
I am not sure I still got the understanding of the concern about the way its implemented. We are using k8s supported way of refreshing bearer tokens which is supported by  caasp generated kube-config.

Now what's suggested might be another approach, does not seems like better approach as there is additional extra step needed to use velum interaction. Whereas in current implemented approach, its works behind the scenes.

Before this, we used to see expired token error when playbook play executed after a day or two and that does not with this fix.

Now having a separate PR is a separate discussion and we have already talked about why its not done in this case and not worth repeating here.

https://kubernetes.io/docs/reference/access-authn-authz/authentication/#option-1-oidc-authenticator

Reply:
I just meant to reuse a little of the code above. If you don't want, fine for me.